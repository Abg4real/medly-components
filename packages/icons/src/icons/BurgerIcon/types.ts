import { IconSizes } from '@medly-components/theme';
import { WithThemeProp } from '@medly-components/utils';

export interface SpanProps extends WithThemeProp {
    size?: IconSizes;
    color?: string;
}

export interface BurgerIconStyledProps extends SpanProps {
    open?: boolean;
}

export interface BurgerIconProps {
    /** Icon size */
    size?: IconSizes;
    /** Icon color */
    color?: string;
    /** Prop to decide its open state */
    open?: boolean;
    /** Function to be called on click on it */
    onClick?: () => void;
}
