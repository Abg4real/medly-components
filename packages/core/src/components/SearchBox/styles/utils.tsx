import { SvgIcon } from '@medly-components/icons';
import { css } from '@medly-components/utils';
import { Props } from '../types';

export const getSearchBoxSize = ({ theme, size }: Props) => theme.searchBox.sizes[size].height;

export const getBorderAndBoxShadow = (color: string, boxShadow: string) => {
    return css`
        border-color: ${color};
        box-shadow: ${boxShadow};
    `;
};

export const getIconWrapperStyle = ({ theme, size }: Props) => {
    return css`
        width: ${getSearchBoxSize};
        min-width: ${getSearchBoxSize};
        height: auto;
        display: inline-flex;
        align-items: center;
        justify-content: center;
        ${SvgIcon} {
            cursor: pointer;
            font-size: ${theme.searchBox.sizes[size].iconSize};
        }
    `;
};
