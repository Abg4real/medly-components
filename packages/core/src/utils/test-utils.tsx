import { defaultTheme } from '@medly-components/theme';
import React from 'react';
import { render, RenderResult } from 'react-testing-library';
import { ThemeProvider } from './styled';

const AllTheProviders: React.FunctionComponent = props => (
    <ThemeProvider theme={defaultTheme}>
        <>{props.children}</>
    </ThemeProvider>
);

const customRender = (ui: React.ReactElement<any>, options?: object): RenderResult => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from 'react-testing-library';

// override render method
export { customRender as render };
