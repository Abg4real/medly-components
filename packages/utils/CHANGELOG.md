# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-beta.16](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.15...@medly-components/utils@1.0.0-beta.16) (2020-04-11)


### Features

* add typography and refactor components based on new typography ([462f2aa](https://github.com/medlypharmacy/medly-components/commit/462f2aa5b24c968b941887e6fcea5fd4e5c7b3f1))





# [1.0.0-beta.15](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.14...@medly-components/utils@1.0.0-beta.15) (2020-04-02)


### Bug Fixes

* **all:** fix lint issues ([ca12080](https://github.com/medlypharmacy/medly-components/commit/ca1208092c7a4d0ce0e22f20cb31c0400c4ddbff))





# [1.0.0-beta.14](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.13...@medly-components/utils@1.0.0-beta.14) (2020-03-30)


### Features

* **all:** add rollup to create bundles ([08b5972](https://github.com/medlypharmacy/medly-components/commit/08b5972f4a221037d14449ffce508f8656d0f447))





# [1.0.0-beta.13](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.12...@medly-components/utils@1.0.0-beta.13) (2020-03-29)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.12](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.11...@medly-components/utils@1.0.0-beta.12) (2020-02-06)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.11](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.10...@medly-components/utils@1.0.0-beta.11) (2020-02-04)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.10](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.9...@medly-components/utils@1.0.0-beta.10) (2020-01-22)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.9](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.8...@medly-components/utils@1.0.0-beta.9) (2020-01-21)


### Features

* **core:** add functionaity to accept date as string also in datepicker component ([3c3c377](https://github.com/medlypharmacy/medly-components/commit/3c3c377a66c470e6765d4a98f0f55b443b5ab0bc))
* **icons:** add bill icon ([2b461c2](https://github.com/medlypharmacy/medly-components/commit/2b461c2478590247498b08f13cad83d225efff6e))





# [1.0.0-beta.8](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.7...@medly-components/utils@1.0.0-beta.8) (2020-01-12)


### Features

* **all:** add side-effects to package.json ([0f78b4c](https://github.com/medlypharmacy/medly-components/commit/0f78b4c2ba432373b9d66861c25a265dacd5d433))





# [1.0.0-beta.7](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.6...@medly-components/utils@1.0.0-beta.7) (2020-01-09)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.6](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.5...@medly-components/utils@1.0.0-beta.6) (2019-12-17)


### Bug Fixes

* **core:** fix modal close issue ([f2d8fe9](https://github.com/medlypharmacy/medly-components/commit/f2d8fe9553f9efdbeb96c8b5b213844f0c472777))





# [1.0.0-beta.5](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.4...@medly-components/utils@1.0.0-beta.5) (2019-12-10)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.4](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.3...@medly-components/utils@1.0.0-beta.4) (2019-11-28)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.3](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.2...@medly-components/utils@1.0.0-beta.3) (2019-11-13)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.2](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.1...@medly-components/utils@1.0.0-beta.2) (2019-11-13)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.1](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.2.0...@medly-components/utils@1.0.0-beta.1) (2019-11-13)


### Bug Fixes

* **all:** remove changelog for pre release setup ([0795f5c](https://github.com/medlypharmacy/medly-components/commit/0795f5c139b529c3f6b65442ff71a8826ecf2c56))
* fixed various issues in date picker ([00fed84](https://github.com/medlypharmacy/medly-components/commit/00fed842b6d363f817de2c8703785c62557cf50d))


### Features

* **all:** remove s3 upload command from all packages ([89ea085](https://github.com/medlypharmacy/medly-components/commit/89ea085082829cba2098342c9793a43e2bc2e24a))
* **core:** add capability to change option in Single select through arrow keys ([fd864b9](https://github.com/medlypharmacy/medly-components/commit/fd864b9eb103bc042086c51236f10222cfc57367))
* added capitalize function to capitalize first letter of the string ([2613604](https://github.com/medlypharmacy/medly-components/commit/261360434dcd02de4cde438350d00696323d16d5))
* added DatePicker and DateRangePicker component ([5dc6d38](https://github.com/medlypharmacy/medly-components/commit/5dc6d38f66f6d1d530f830da13a9624715f6f704))
