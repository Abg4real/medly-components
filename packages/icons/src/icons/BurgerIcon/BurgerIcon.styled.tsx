import { defaultTheme } from '@medly-components/theme';
import { css, styled } from '@medly-components/utils';
import { BurgerIconStyledProps, SpanProps } from './types';

const getSize = ({ theme, size }: BurgerIconStyledProps) => theme.icon.sizes[size];

export const SpanStyled = styled('span')<SpanProps>`
    display: block;
    position: absolute;
    height: calc(${getSize} * 0.2);
    width: 100%;
    background: ${({ theme, color }) => color || theme.icon.defaultColor};
    border-radius: calc(${getSize} * 0.2);
    opacity: 1;
    left: 0;
    transform: rotate(0deg);
    transition: 0.25s ease-in-out;

    &:nth-child(1) {
        top: 0px;
    }

    &:nth-child(2) {
        top: calc(${getSize} * 0.4);
    }

    &:nth-child(3) {
        top: calc(${getSize} * 0.8);
    }
`;
SpanStyled.defaultProps = {
    theme: defaultTheme
};

const isOpen = () => css<BurgerIconStyledProps>`
    ${SpanStyled} {
        &:nth-child(1) {
            top: calc(${getSize} * 0.4);
            transform: rotate(135deg);
        }

        &:nth-child(2) {
            opacity: 0;
            left: -60px;
        }

        &:nth-child(3) {
            top: calc(${getSize} * 0.4);
            transform: rotate(-135deg);
        }
    }
`;

export const BurgerIconStyled = styled('div')<BurgerIconStyledProps>`
    display: inline-block;
    width: calc(${getSize} * 1.33);
    height: ${getSize};
    position: relative;
    margin: 10px;
    transform: rotate(0deg);
    transition: 0.2s ease-in-out;
    cursor: pointer;

    ${({ open }) => open && isOpen()}
`;

BurgerIconStyled.defaultProps = {
    theme: defaultTheme
};
