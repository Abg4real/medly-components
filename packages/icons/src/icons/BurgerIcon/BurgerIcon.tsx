import { WithStyle } from '@medly-components/utils';
import React, { useState } from 'react';
import { BurgerIconStyled, SpanStyled } from './BurgerIcon.styled';
import { BurgerIconProps } from './types';

export const BurgerIcon: React.SFC<BurgerIconProps> & WithStyle = React.memo(props => {
    const [open, setOpenState] = useState(false);
    const { size, color } = props;

    const onclickHandler = () => {
        props.onClick && props.onClick(!open);
        setOpenState(!open);
    };

    return (
        <BurgerIconStyled id="burgerIcon" {...{ size, color, open, onClick: onclickHandler }}>
            <SpanStyled {...{ size, color }} />
            <SpanStyled {...{ size, color }} />
            <SpanStyled {...{ size, color }} />
        </BurgerIconStyled>
    );
});
BurgerIcon.displayName = 'BurgerIcon';
BurgerIcon.defaultProps = {
    size: 'L'
};
BurgerIcon.Style = BurgerIconStyled;

export default BurgerIcon;
