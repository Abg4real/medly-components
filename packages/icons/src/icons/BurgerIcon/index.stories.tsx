import { defaultTheme, IconSizes } from '@medly-components/theme';
import { color, select } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import React from 'react';
import { BurgerIcon } from './BurgerIcon';

const size: IconSizes[] = ['XS', 'S', 'M', 'L', 'XL'];
const defaultColor = defaultTheme.icon.defaultColor;

storiesOf('Icons', module).add('Burger', () => <BurgerIcon size={select('Size', size, 'S')} color={color('Color', defaultColor)} />);
