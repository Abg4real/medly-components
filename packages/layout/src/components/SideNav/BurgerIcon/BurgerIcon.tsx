import { WithStyle } from '@medly-components/utils';
import React, { SFC } from 'react';
import { BurgerIconStyled, SpanStyled } from './BurgerIcon.styled';
import { BurgerIconProps } from './types';

export const BurgerIcon: SFC<BurgerIconProps> & WithStyle = React.memo(props => {
    const { size, color } = props;
    return (
        <BurgerIconStyled id="burgerIcon" {...props}>
            <SpanStyled {...{ size, color }} />
            <SpanStyled {...{ size, color }} />
            <SpanStyled {...{ size, color }} />
        </BurgerIconStyled>
    );
});
BurgerIcon.displayName = 'BurgerIcon';
BurgerIcon.defaultProps = {
    size: 'L',
    open: false
};
BurgerIcon.Style = BurgerIconStyled;
