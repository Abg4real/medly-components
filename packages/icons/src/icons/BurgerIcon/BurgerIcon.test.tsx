import { TestUtils } from '@medly-components/utils';
import 'jest-styled-components';
import React from 'react';
import BurgerIcon from './BurgerIcon';

const { render, fireEvent, cleanup } = TestUtils;
afterEach(cleanup);

describe('BurgerIcon component', () => {
    it('should render with default props', () => {
        const { container } = render(<BurgerIcon />);
        expect(container).toMatchSnapshot();
    });

    it('should render with all props given', () => {
        const { container } = render(<BurgerIcon size="M" color="pink" />);
        expect(container).toMatchSnapshot();
    });

    it('should call onClick prop with expected open state', () => {
        const mockOnClick = jest.fn();
        const { container } = render(<BurgerIcon size="M" color="pink" onClick={mockOnClick} />);
        fireEvent.click(container.querySelector('button'));
        expect(mockOnClick).toHaveBeenCalledWith(true);
        fireEvent.click(container.querySelector('button'));
        expect(mockOnClick).toHaveBeenCalledWith(false);
    });
});
