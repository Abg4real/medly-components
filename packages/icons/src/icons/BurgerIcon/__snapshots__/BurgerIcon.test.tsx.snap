// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BurgerIcon component should render with all props given 1`] = `
.c1 {
  display: block;
  position: absolute;
  width: calc(24px * 1.33);
  height: calc(24px * 0.2);
  background: pink;
  border-radius: calc(24px * 0.2);
  opacity: 1;
  left: 0;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: 0.25s ease-in-out;
  transition: 0.25s ease-in-out;
}

.c1:nth-child(1) {
  top: 0px;
}

.c1:nth-child(2) {
  top: calc(24px * 0.4);
}

.c1:nth-child(3) {
  top: calc(24px * 0.8);
}

.c0 {
  border: none;
  background-color: transparent;
  outline: none;
  width: calc(24px * 1.33);
  height: 24px;
  position: relative;
  margin: 10px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: 0.2s ease-in-out;
  transition: 0.2s ease-in-out;
  cursor: pointer;
}

<div>
  <button
    class="c0"
    color="pink"
    id="burgerIcon"
  >
    <span
      class="c1"
      color="pink"
    />
    <span
      class="c1"
      color="pink"
    />
    <span
      class="c1"
      color="pink"
    />
  </button>
</div>
`;

exports[`BurgerIcon component should render with default props 1`] = `
.c1 {
  display: block;
  position: absolute;
  width: calc(36px * 1.33);
  height: calc(36px * 0.2);
  background: #555;
  border-radius: calc(36px * 0.2);
  opacity: 1;
  left: 0;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: 0.25s ease-in-out;
  transition: 0.25s ease-in-out;
}

.c1:nth-child(1) {
  top: 0px;
}

.c1:nth-child(2) {
  top: calc(36px * 0.4);
}

.c1:nth-child(3) {
  top: calc(36px * 0.8);
}

.c0 {
  border: none;
  background-color: transparent;
  outline: none;
  width: calc(36px * 1.33);
  height: 36px;
  position: relative;
  margin: 10px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: 0.2s ease-in-out;
  transition: 0.2s ease-in-out;
  cursor: pointer;
}

<div>
  <button
    class="c0"
    id="burgerIcon"
  >
    <span
      class="c1"
    />
    <span
      class="c1"
    />
    <span
      class="c1"
    />
  </button>
</div>
`;
